ci/setup_env.sh

```diff
@@ -0,0 +1,125 @@
+
+if [ -z "${TRAVIS_TAG}" ]; then
+  echo "TRAVIS_TAG:                           No TAG"
+else
+  echo "TRAVIS_TAG:                           ${TRAVIS_TAG}"
+fi
+echo "package version:                      ${package_version}"
+
+# Functions
+function import_gpg_keys() {
+  # shellcheck disable=SC2145
+  printf "%s\n" "Tagged build, fetching keys:" "${@}" ""
+  # shellcheck disable=SC2207
+  declare -r my_arr=( $(echo "${@}" | tr " " "\n") )
+
+  for key in "${my_arr[@]}"; do
+    echo "Importing key: ${key}"
+    gpg -v --batch --keyserver hkps://keys.openpgp.org --recv-keys "${key}" ||
+    gpg -v --batch --keyserver hkp://keyserver.ubuntu.com --recv-keys "${key}" ||
+    gpg -v --batch --keyserver hkp://pgp.mit.edu:80 --recv-keys "${key}" ||
+    gpg -v --batch --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys "${key}"
+  done
+}
+
+function check_signed_tag() {
+  # Checking if git tag signed by the maintainers
+  if git verify-tag -v "${1}"; then
+    echo "${1} is a valid signed tag"
+  else
+    echo "Git tag's = ${1} gpg signature is NOT valid. The build is not going to be released..."
+  fi
+}
+
+# empty key.asc file in case we're not signing
+touch "${HOME}/key.asc"
+
+if [ -n "${TRAVIS_TAG}" ]; then
+  # checking if MAINTAINER_KEYS is set
+  if [ -z "${MAINTAINER_KEYS}" ]; then
+    echo "MAINTAINER_KEYS variable is not set. Make sure to set it up for release build!!!"
+    exit 1
+  fi
+
+  # shellcheck disable=SC2155
+  export GNUPGHOME="$(mktemp -d 2>/dev/null || mktemp -d -t 'GNUPGHOME')"
+
+  # Checking PROD vs DEV build
+  echo "The current production branch is: ${PROD_RELEASE_BRANCH}"
+  echo "The current development branch is: ${DEV_RELEASE_BRANCH}"
+
+  import_gpg_keys "${MAINTAINER_KEYS}"
+
+  # Checking git tag gpg signature requirement
+  if ( check_signed_tag "${TRAVIS_TAG}" ); then
+    # Prod vs dev release
+    if ( git branch -r --contains "${TRAVIS_TAG}" | grep -xqE ". origin\/${PROD_RELEASE_BRANCH}$" ); then
+      # Checking if package version matches PROD release version
+      if ! [[ "${package_version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-RC[0-9]+)?$ ]]; then
+        echo "Aborting, package version is in the wrong format."
+        exit 1
+      fi
+
+      # Checking Github tag format
+      if ! [[ "${TRAVIS_TAG}" == "${package_version}" ]]; then
+        echo "Aborting, tag format differs from the version under build.sbt file."
+        exit 1
+      fi
+
+      # Announcing PROD release
+      echo "" && echo "=== Production release build ===" && echo ""
+      echo "Fetching maven gpg signing keys."
+      curl -sLH "Authorization: token ${GITHUB_TOKEN}" -H "Accept: application/vnd.github.v3.raw" "${MAVEN_KEY_ARCHIVE_URL}" |
+        openssl enc -d -aes-256-cbc -md sha256 -pass pass:"${MAVEN_KEY_ARCHIVE_PASSWORD}" |
+        tar -xzf- -C "${HOME}"
+      export CONTAINER_PUBLISH="true"
+    elif ( git branch -r --contains "${TRAVIS_TAG}" | grep -xqE ". origin\/${DEV_RELEASE_BRANCH}$" ); then
```

Comment:
Do we allow to publish snapshots only from `dev` branch?
If I'm not wrong on SDK we allow to create a snapshot from whatever branch.
Can be much more useful, considering that pushing directly to `dev` is restricted, so in any case we are obliged to create a tag in a third branch and then merge via PR.

Reply:
Yes, @i-Alex. Currently SNAPSHOTs are allowed to be published only off `dev` branch. We can change this behavior and allow publishing off any branch. @cronicc what do you think?

Reply:
@ysibirski I'm ok with relaxing the restriction for SNAPSHOT releases, but please keep the protection for production releases to be only published from the main branch.

Reply:
Fixed